tuf/api/metadata.py

```diff
@@ -1258,6 +1258,64 @@ def to_dict(self) -> Dict[str, Any]:
         return snapshot_dict
 
 
+class HashDelegations:
+    """A container with information about a succinct hash delegation.
```

Comment:
I think I like this: I imagined it would be OO-overkill but maybe not: 

We could move some functionality here 
* like the delegation math you need, see other comment: I think it'll look just fine
* but also the math required to create new succinct delegations -- where @kairoaraujo can say something like _"I want 2**14 bins, give me the delegation but also give me a generator for all the bin names so I can create the metadata for those"._

We'll want to be careful with the name though, there are so many delegation related names. We now have three completely different names for the same concept, this does not feel correct: 
* spec calls it `succinct_hash_delegations`
* DelegatedRole calls it `succinct_hash_info`
* here it's called `HashDelegations`

Reply:
I agree about the naming.
Honestly I wasn't sure how to call it at all.
Maybe `SuccinctHashInfo`? But I don't like the suffix `info` as it doesn't bring any new context...
On another hand, `SuccinctHashDelegations` becomes too much...

Reply:
On the other hand, class name is not used often so that could still work if you can keep the variable name as something clearly related but a little shorter, something like:

succinct_delegations = SuccinctHashDelegations(3, "bins")

Admittedly that is a bit longer than your current variable name